packages/open-payments/src/types.ts

```diff
@@ -18,6 +18,20 @@ export type ILPStreamConnection =
 export type OutgoingPayment = RSComponents['schemas']['outgoing-payment']
 export type CreateOutgoingPaymentArgs =
   RSOperations['create-outgoing-payment']['requestBody']['content']['application/json']
+type PaginationResult<T> = {
+  pagination: RSComponents['schemas']['page-info']
+  result: T[]
+}
+export type OutgoingPaymentPaginationResult = PaginationResult<OutgoingPayment>
+export type ForwardPagination =
+  RSComponents['schemas']['forward-pagination'] & {
+    last?: never
```

Comment:
I was surprised this was necessary because we have `additionalProperties: false`
https://github.com/interledger/open-payments/blob/main/openapi/resource-server.yaml#L1214

Would strict type checking make the `never` unnecessary?
If so, would that be something we couldn't enforce/guarantee on clients of this library?

Reply:
So this isn't an Open API validation issue, this is for the TS types. Without the `never` you can provide `first` `last` and `cursor` and TS won't complain:

<img width="269" alt="Screen Shot 2022-12-13 at 11 26 15 AM" src="https://user-images.githubusercontent.com/15069181/207293270-c3b08547-d9c5-4c42-8c02-95d3caa307c9.png">

while with `never`, it'll be an error:

<img width="438" alt="Screen Shot 2022-12-13 at 11 26 30 AM" src="https://user-images.githubusercontent.com/15069181/207293377-4b7dab9b-673d-4e54-8f65-e2c75d5772f8.png">


https://stackoverflow.com/a/61350902

Reply:
Right, I'm just realizing/thinking through (and I guess trying to avoid) the implications of `additionalProperties: false` not being enforced in the generated types without specifying `never`s.

Reply:
Isn't that mostly a concern for these `oneOf` union types?

Reply:
Yeah, I don't think there's any concerns here with the OpenAPI validation itself.
But I hadn't considered that a generated type from an object with `additionalProperties: false` could accept an object with additional properties (unless the specified properties are manual defined as `never`)